Adafruit_ILI9341::scrollTo uint16 vs uint8

while compiling with all warnings on:
/home/merlin/Arduino/libraries/Adafruit_ILI9341/Adafruit_ILI9341.cpp:259:46: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
     sendCommand(ILI9341_VSCRSADD, (uint8_t*) y, 2);

Sure enough, the function is
void Adafruit_ILI9341::scrollTo(uint16_t y) {
    sendCommand(ILI9341_VSCRSADD, (uint8_t*) y, 2);
}

Given that a TFT can be wider than 256 chars, uint16_t seems reasonable, but spiWrite(*dataBytes) may only allow a single byte, i.e. a scroll of up to 255 pixels?

If so, changing scrollTo to take uint8_t seems reasonable?

--- a/Adafruit_ILI9341.cpp
+++ b/Adafruit_ILI9341.cpp
@@ -256,7 +256,10 @@ void Adafruit_ILI9341::invertDisplay(bool invert) {
 */
 /**************************************************************************/
 void Adafruit_ILI9341::scrollTo(uint16_t y) {
-    sendCommand(ILI9341_VSCRSADD, (uint8_t*) y, 2);
+    uint8_t data[2];
+    data[0] = y >> 8;
+    data[1] = y && 0xff;
+    sendCommand(ILI9341_VSCRSADD, (uint8_t*) data, 2);
 }
 
 /**************************************************************************/

